installers/vagrant-dev/radiasoft-download.sh

```diff
@@ -171,18 +187,25 @@ EOF
     fi
     vagrant ssh <<EOF
 $(install_vars_export)
-curl $(install_depot_server)/index.sh | bash -s redhat-dev
+curl $(install_depot_server)/index.sh | \
+  vagrant_dev_ignore_git_dir_ownership=$(vagrant_dev_ignore_git_dir_ownership $os) \
+  bash -s redhat-dev
 EOF
     vagrant_dev_post_install
 }
 
 vagrant_dev_mounts() {
+    declare first="${1:+1}"
```

Comment:
I don't think the quotes are necessary. And I would think we'd want to abort if the parameter wasn't passed. This is an internal function.

Reply:
Can you elaborate on why no to use quotes? I get that the value in $1 has no spaces in it so quotes aren't strictly necessary. But, in my mind that creates more tightly coupled code. We are coupling the fact that the value has no quotes to how we are using it here and that coupling isn't necessary. 

Somewhere along the way we need to set a default for $first. Otherwise bash will complain about an unbound variable.

```bash
a() {
    b '1'
    b ''
}

b() {
    declare first=$1
    x=$(c $first)
    echo "$x"
}

c() {
    declare first=$1
    echo "first=$first"
}

a
```
```
$ bash x.sh
first=1
x.sh: line 15: $1: unbound variable
```
 
So should we set the default when calling? Is there a reason to prefer handling default in the caller vs the callee?

Reply:
The error that causes the unbound variable is on line 8, it should be:

```sh
    x=$(c "$first")
```

Then, the script runs correctly.  You need to quote in command expansions (`c $first`) and array values, e.g. 

```sh
$ x=( $(seq 3) ) y=( "$(seq 3)" )
$ (IFS=,; echo "${x[*]}" "${y[*]}")
1,2,3 1
2
3
```

This is the only expansion that happens in assignments:

> Each variable assignment is checked for unquoted tilde-prefixes immediately following a ":" or "=". In these cases, tilde expansion is also performed. Consequently, one may use file names with tildes in assignments to PATH, MAILPATH, and CDPATH, and the shell assigns the expanded value.

I asked ChatGPT, and it seems it didn't know this. It does

Reply:
Thanks.

If I wanted to quote everything is this right? `x="$(c "$first")"`?

Reply:
Yes, that's the right way to do it, and typically used elsewhere, e.g.

```sh
echo "$(c "$first")"
```

The quotes inside of subshell expansions is why subshell expansions were introduced. This doesn't work in backticks.

Reply:
Thanks for walking me through it b2d113e